toolchain/check/testdata/impl/no_prelude/impl_assoc_const.carbon

```diff
@@ -80,6 +80,56 @@ impl () as I4 where .T4 = BAD1 and .T5 = {.a: {}} and .T6 = BAD2;
 // CHECK:STDERR:
 impl () as I4 where .T4 = {.b: {}} and .T5 = BAD3 and .T6 = BAD4 {}
 
+// --- fail_mismatch_mismatch_value.carbon
+library "[[@TEST_NAME]]";
+
+interface I5 {
+  let T7:! type;
+  let T8:! type;
+  let T9:! type;
+}
+
+class B1 { }
+class B2 { }
+class B3 { }
+class B4 { }
+class B5 { }
+class B6 { }
+class B7 { }
+class B8 { }
+class G1 { }
+class G2 { }
+
+// This is testing that it won't complain about mismatching values between
+// impl redeclarations if one of them is an error. We use mismatching values
+// for an associated constant within a single impl declaration in order to
+// create an error that only affects that member instead of the whole facet
+// type.
+
+// CHECK:STDERR: fail_mismatch_mismatch_value.carbon:[[@LINE+8]]:12: error: associated constant T9 given two different values [AssociatedConstantWithDifferentValues]
+// CHECK:STDERR: impl () as I5 where .T7 = B1 and .T7 = B2 and .T8 = G1 and .T9 = B3 and .T9 = B4;
+// CHECK:STDERR:            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+// CHECK:STDERR:
+// CHECK:STDERR: fail_mismatch_mismatch_value.carbon:[[@LINE+4]]:12: error: associated constant T7 given two different values [AssociatedConstantWithDifferentValues]
+// CHECK:STDERR: impl () as I5 where .T7 = B1 and .T7 = B2 and .T8 = G1 and .T9 = B3 and .T9 = B4;
+// CHECK:STDERR:            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+// CHECK:STDERR:
+impl () as I5 where .T7 = B1 and .T7 = B2 and .T8 = G1 and .T9 = B3 and .T9 = B4;
+
+// CHECK:STDERR: fail_mismatch_mismatch_value.carbon:[[@LINE+12]]:12: error: associated constant T9 given two different values [AssociatedConstantWithDifferentValues]
+// CHECK:STDERR: impl () as I5 where .T7 = G2 and .T8 = B5 and .T8 = B6 and .T9 = B7 and .T9 = B8 { }
+// CHECK:STDERR:            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+// CHECK:STDERR:
+// CHECK:STDERR: fail_mismatch_mismatch_value.carbon:[[@LINE+8]]:12: error: associated constant T8 given two different values [AssociatedConstantWithDifferentValues]
+// CHECK:STDERR: impl () as I5 where .T7 = G2 and .T8 = B5 and .T8 = B6 and .T9 = B7 and .T9 = B8 { }
+// CHECK:STDERR:            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+// CHECK:STDERR:
+// CHECK:STDERR: fail_mismatch_mismatch_value.carbon:[[@LINE-10]]:1: error: impl declared but not defined [ImplMissingDefinition]
+// CHECK:STDERR: impl () as I5 where .T7 = B1 and .T7 = B2 and .T8 = G1 and .T9 = B3 and .T9 = B4;
+// CHECK:STDERR: ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+// CHECK:STDERR:
```

Comment:
The comment above says that this was testing impl redeclarations, but this error suggests we don't treat these as impl redeclarations. Was there supposed to be an impl redeclaration here, or is the comment wrong / am I misreading it?

Reply:
The rules have changed since I wrote this. Previously impl redeclarations allowed syntactic mismatches in the `where` part. I think I'm going to just delete this test.